compiler/rustc_log/src/lib.rs

```diff
@@ -155,18 +155,19 @@ where
         Err(_) => {} // no wraptree
     }
 
-    let subscriber = build_subscriber().with(layer.with_filter(filter));
+    let subscriber = build_subscriber();
+    // NOTE: It is important to make sure that the filter is applied on the last layer
     match cfg.backtrace {
         Ok(backtrace_target) => {
             let fmt_layer = tracing_subscriber::fmt::layer()
                 .with_writer(io::stderr)
                 .without_time()
                 .event_format(BacktraceFormatter { backtrace_target });
-            let subscriber = subscriber.with(fmt_layer);
+            let subscriber = subscriber.with(layer).with(fmt_layer).with(filter);
```

Comment:
This applies the filter to *all* layers. I think that's what breaks Miri, which is adding its own layer that needs to have a *different* filter.